internal/command/user_v2_human.go

```diff
@@ -0,0 +1,559 @@
+	}
+
+	if human.ID == "" {
+		human.ID, err = c.idGenerator.Next()
+		if err != nil {
+			return err
+		}
+	}
+
+	// only check if user is already existing
+	existingHuman, err := c.userExistsWriteModel(
+		ctx,
+		human.ID,
+		resourceOwner,
+	)
+	if err != nil {
+		return err
+	}
+	if isUserStateExists(existingHuman.UserState) {
+		return zerrors.ThrowPreconditionFailed(nil, "COMMAND-7yiox1isql", "Errors.User.AlreadyExisting")
+	}
+
+	domainPolicy, err := c.domainPolicyWriteModel(ctx, resourceOwner)
+	if err != nil {
+		return err
+	}
+
+	if err = c.userValidateDomain(ctx, resourceOwner, human.Username, domainPolicy.UserLoginMustBeDomain); err != nil {
+		return err
+	}
+	var createCmd humanCreationCommand
+	if human.Register {
+		createCmd = user.NewHumanRegisteredEvent(
+			ctx,
+			&existingHuman.Aggregate().Aggregate,
+			human.Username,
+			human.FirstName,
+			human.LastName,
+			human.NickName,
+			human.DisplayName,
+			human.PreferredLanguage,
+			human.Gender,
+			human.Email.Address,
+			domainPolicy.UserLoginMustBeDomain,
+		)
+	} else {
+		createCmd = user.NewHumanAddedEvent(
+			ctx,
+			&existingHuman.Aggregate().Aggregate,
+			human.Username,
+			human.FirstName,
+			human.LastName,
+			human.NickName,
+			human.DisplayName,
+			human.PreferredLanguage,
+			human.Gender,
+			human.Email.Address,
+			domainPolicy.UserLoginMustBeDomain,
+		)
+	}
+
+	if human.Phone.Number != "" {
+		createCmd.AddPhoneData(human.Phone.Number)
+	}
+
+	// separated to change when old user logic is not used anymore
+	filter := c.eventstore.Filter //nolint:staticcheck
+	if err := addHumanCommandPassword(ctx, filter, createCmd, human, c.userPasswordHasher); err != nil {
+		return err
+	}
+
+	cmds := make([]eventstore.Command, 0, 3)
+	cmds = append(cmds, createCmd)
+
+	cmds, err = c.addHumanCommandEmail(ctx, filter, cmds, existingHuman.Aggregate(), human, alg, allowInitMail)
+	if err != nil {
+		return err
+	}
+
+	cmds, err = c.addHumanCommandPhone(ctx, filter, cmds, existingHuman.Aggregate(), human, alg)
+	if err != nil {
+		return err
+	}
+
+	for _, metadataEntry := range human.Metadata {
+		cmds = append(cmds, user.NewMetadataSetEvent(
+			ctx,
+			&existingHuman.Aggregate().Aggregate,
+			metadataEntry.Key,
+			metadataEntry.Value,
+		))
+	}
+	for _, link := range human.Links {
+		cmd, err := addLink(ctx, filter, existingHuman.Aggregate(), link)
+		if err != nil {
+			return err
+		}
+		cmds = append(cmds, cmd)
+	}
+
+	if len(cmds) == 0 {
+		human.Details = writeModelToObjectDetails(&existingHuman.WriteModel)
+		return nil
+	}
+
+	err = c.pushAppendAndReduce(ctx, existingHuman, cmds...)
+	if err != nil {
+		return err
+	}
+	human.Details = writeModelToObjectDetails(&existingHuman.WriteModel)
+	return nil
+}
+
+func (c *Commands) LockUserHuman(ctx context.Context, resourceOwner string, userID string) (*domain.ObjectDetails, error) {
+	if resourceOwner == "" {
+		return nil, zerrors.ThrowInvalidArgument(nil, "COMMA-u3pf6vqrub", "Errors.Internal")
+	}
```

Comment:
as already mentioned, these functions need to be renamed and the resourceOwner check will not work

Reply:
moved to a different file to separate it from human